### PR TITLE
scoop up the right pre elements' text content

### LINKED ISSUE #6161: Discrepancy between w.d.m.o and d.m.o for embedded live examples with hidden fragments when opening in CodePen / JSFiddle
**Summary**
When a live sample has hidden parts (cf. the first live sample of https://wiki.developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Auto-placement_in_CSS_Grid_Layout ), clicking "Open in CodePen" (or "Open in JSFiddle") does not produce the same result whether we click from developer.mozilla.org ([this page](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Auto-placement_in_CSS_Grid_Layout)) or from wiki.developer.mozilla.org ([this page](https://wiki.developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Auto-placement_in_CSS_Grid_Layout)).

**Steps To Reproduce (STR)**

1. Open https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Auto-placement_in_CSS_Grid_Layout
2. Click on the first "Open in CodePen" button

**Actual behavior**

The embedded pen (in CodePen) does not take into account the `<pre>`s that had `class="hidden"`.
Result from opening from d.m.o on CodePen: https://codepen.io/SphinxKnight/pen/jOOjwLG

**Expected behavior**

* The result in CodePen / JSFiddle should be the same than the one on the MDN page
* Result from opening from w.d.m.o on CodePen: https://codepen.io/SphinxKnight/pen/gOONRGM

* There should not be any discrepancy for live samples between d.m.o and w.d.m.o

**Additional context**
* Reported earlier through https://github.com/mdn/sprints/issues/2376
* Macro: https://github.com/mdn/kumascript/blob/master/macros/EmbedLiveSample.ejs
* Kuma: https://github.com/mdn/kuma/blob/master/kuma/javascript/src/live-examples.js

### PR DESCRIPTION
**Pull request by peterbe**
Fixes #6161

Jump straight to [this comment](https://github.com/mdn/kuma/issues/6161#issuecomment-561805170) to see what the problem is. The [next comment](https://github.com/mdn/kuma/issues/6161#issuecomment-561805517) demonstrates the solution.

### CODE REVIEW

File: kuma/javascript/src/live-examples.js [hunk 1]
```diff
@@ -48,20 +48,14 @@ export function addLiveExampleButtons(rootElement) {
...
+        let css = section.querySelectorAll('pre[class*=css]');
+        let js = section.querySelectorAll('pre[class*=js]');
 
         // Now get the source code out of those pre elements
-        let htmlCode = html ? html.textContent : '';
-        let jsCode = js ? js.textContent : '';
-        let cssCode = css ? css.textContent : '';
+        let htmlCode = [...html].map(p => p.textContent).join('');
+        let jsCode = [...js].map(p => p.textContent).join('');
+        let cssCode = [...css].map(p => p.textContent).join('');
```
Comment (schalkneethling) [inline kuma/javascript/src/live-examples.js]: This is so clean and nice to read, my only concern is IE :( Perhaps doing it more IE friendly would be the better option, thanks @peterbe

Comment (peterbe) [reply kuma/javascript/src/live-examples.js]: I don't get it. Why wouldn't this work in IE?

Comment (schalkneethling) [reply kuma/javascript/src/live-examples.js]: The spread operator is not supported in IE - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Browser_compatibility

and neither is arrow functions - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#Browser_compatibility

Perhaps I am not thinking about it correctly?

Comment (peterbe) [reply kuma/javascript/src/live-examples.js]: Hehe. You forgot that this is converted to ES5  by Babel :)

Comment (schalkneethling) [reply kuma/javascript/src/live-examples.js]: Doh! For some reason, I had it in my head that this was part of the old JS :p

**Pull request merged**